src/app/contexts/EventContext/useWindowEvent/index.test.js

```diff
@@ -0,0 +1,36 @@
+import { useEffect } from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import useWindowEvent from './index';
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('useWindowEvent', () => {
+  const fn1 = jest.fn();
+
+  it('should trigger window click handler', () => {
+    useWindowEvent('click', fn1, false)();
+
+    window.dispatchEvent(new Event('click'));
+
+    expect(fn1).toHaveBeenCalled();
+  });
+
+  it('should cleanup', () => {
+    const fn2 = jest.fn();
+    const { unmount } = renderHook(() => {
+      useEffect(useWindowEvent('click', fn2, false));
+    });
+
+    window.dispatchEvent(new Event('click'));
+
+    expect(fn2).toHaveBeenCalledTimes(1);
+
+    unmount();
+
+    window.dispatchEvent(new Event('click'));
+
+    expect(fn2).toHaveBeenCalledTimes(1);
```

Comment:
I might be wrong, but I would have expected this to be called 0 times

Reply:
As explained on slack,

```
- The fn2 function was bound to the click event on the window.
- And that’s verified in the previous line, by making sure it’s called once.
- We then unmount the component, and trigger the click even on the window again.
- The number of times that fn2 was called should still be once, unless we reset the mocks, which we didn’t.
```

Reply:
ah ok thanks for explaining